lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -899,35 +899,132 @@ class MongoClientInterface {
             return cb(errors.InternalError);
         });
     }
-
     /**
-     * Put object when versioning is not enabled
-     * @param {Object} c bucket collection
-     * @param {String} bucketName bucket name
-     * @param {String} objName object name
-     * @param {Object} objVal object metadata
-     * @param {Object} params params
-     * @param {Object} log logger
-     * @param {Function} cb callback
-     * @return {undefined}
+     * Puts an object into a MongoDB collection.
+     * Depending on the parameters, the object is either directly put into the collection
+     * or the existing object is marked as deleted and a new object is inserted.
+     *
+     * @param {Object} collection - The MongoDB collection to put the object into.
+     * @param {string} bucketName - The name of the bucket the object belongs to.
+     * @param {string} objName - The name of the object.
+     * @param {Object} value - The value of the object.
+     * @param {Object} params - Additional parameters.
+     * @param {string} params.vFormat - object key format.
+     * @param {boolean} params.needOplogUpdate - If true, the object is directly put into the collection
+     * with updating the operation log.
+     * @param {Object} log - The logger to use.
+     * @param {Function} cb - The callback function to call when the operation is complete. It is called with an error
+     * if there is an issue with the operation.
+     * @returns {Promise} A promise that resolves when the operation is complete. The promise is rejected with an error
+     * if there is an issue with the operation.
      */
-    putObjectNoVer(c, bucketName, objName, objVal, params, log, cb) {
-        const masterKey = formatMasterKey(objName, params.vFormat);
-        c.updateOne({
-            _id: masterKey,
-        }, {
+    putObjectNoVer(collection, bucketName, objName, value, params, log, cb) {
+        if (params?.needOplogUpdate) {
+            return this.putObjectNoVerWithOplogUpdate(collection, bucketName, objName, value, params, log, cb);
+        }
+        const key = formatMasterKey(objName, params.vFormat);
+        const putFilter = { _id: key };
+        return collection.updateOne(putFilter, {
             $set: {
-                _id: masterKey,
-                value: objVal,
+                _id: key,
+                value,
             },
         }, {
             upsert: true,
-        }).then(() => cb()).catch((err) => {
-            log.error('putObjectNoVer: error putting obect with no versioning', { error: err.message });
+        }).then(() => cb()).catch(err => {
+            log.error('internalPutObject: error putting object',
+                { bucket: bucketName, object: key, error: err.message });
             return cb(errors.InternalError);
```

Comment:
probably no point changing this, old code works just as well ?
(and `putObjectNoVer: error putting obect with no versioning` error message was more appropriate, giving details about which part of the code fails...)

Reply:
code was kept as is, the error message is fixed here : https://github.com/scality/Arsenal/pull/2249/commits/043b64c3d17ad2bbcc3ea34ddc2b3f6590c436b1#diff-9c207befbd8f4d79a3b6e3d7d143444e804ca1684236b8fd9fa0f9e03c53477aR935